src/java.base/share/classes/jdk/internal/jimage/ImageBufferCache.java

```diff
@@ -39,12 +41,31 @@
 class ImageBufferCache {
     private static final int MAX_CACHED_BUFFERS = 3;
     private static final int LARGE_BUFFER = 0x10000;
-    private static final ThreadLocal<BufferReference[]> CACHE =
-        new ThreadLocal<BufferReference[]>() {
+
+    /*
+     * We used to have a class BufferReference extending from WeakReference<ByteBuffer>.
```

Comment:
I think this comment needs to be re-worded to drop "We used to have ...", "Solution is ..." and the other history. Instead it should provide a clear paragraph that explains the value of the TL for anyone reading this code.